helios-services/src/main/java/com/spotify/helios/rollingupdate/RollingUndeployPlanner.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2014 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.helios.rollingupdate;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.spotify.helios.common.descriptors.DeploymentGroup;
+import com.spotify.helios.common.descriptors.HostStatus;
+import com.spotify.helios.common.descriptors.RolloutTask;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class RollingUndeployPlanner implements RolloutPlanner {
+
+  private final DeploymentGroup deploymentGroup;
+
+  private RollingUndeployPlanner(final DeploymentGroup deploymentGroup) {
+    this.deploymentGroup = checkNotNull(deploymentGroup, "deploymentGroup");
+  }
+
+  public static RollingUndeployPlanner of(final DeploymentGroup deploymentGroup) {
+    return new RollingUndeployPlanner(deploymentGroup);
+  }
+
+  @Override
+  public List<RolloutTask> plan(final Map<String, HostStatus> hostsAndStatuses) {
+    // we only care about hosts that are UP
+    final List<String> hosts = Lists.newArrayList();
+    for (final Map.Entry<String, HostStatus> entry : hostsAndStatuses.entrySet()) {
+      if (entry.getValue().getStatus().equals(HostStatus.Status.UP)) {
+        hosts.add(entry.getKey());
+      }
+    }
```

Comment:
YMMV on whether this is more or less readable, but this could also be 

```java
final List<String> hosts = hostsAndStatuses.entrySet().stream()
  // we only care about hosts that are UP
  .filter(entry -> entry.getValue.getStatus() == HostStatus.Status.UP)
  .map(entry -> entry.getKey())
  .collect(Collectors.toList());
```